### PR TITLE
Creating the entities needed for checkout

### PR DESCRIPTION
**Pull request by mikethetike**
This PR creates all the entities that are needed for the checkout process so that future PRs can use them to implement the API methods.

Some of the `add_to_cart` kind of tests have been commented out because the logic will change with the new structure.

### CODE REVIEW

**Comment by Krakaw** [comment]
Looks good to me, only thing I saw was in tariclient.rs your test data the expire_date is set to Sun Apr 26 19:46:39 SAST 1970

**Comment by mikethetike** [comment]
Thanks, I'll fix that when we have actual data going through it

File: db/migrations/00000000000025_assets/up.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE assets (
```
Comment (SWvheerden) [inline db/migrations/00000000000025_assets/up.sql]: currently the assets have a uuid for the database, but they have unique string id as the assit_id which is used on the communication outward. 

Perhaps store both, or use the string as primary key?

File: db/migrations/00000000000029_ticket_instances/up.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE ticket_instances (
+  id uuid PRIMARY KEY DEFAULT gen_random_uuid() NOT NULL,
+  asset_id uuid NOT NULL REFERENCES assets(id),
```
Comment (SWvheerden) [inline db/migrations/00000000000029_ticket_instances/up.sql]: see comment on assits, but here it makes sense to link to the id of the asset table

File: db/src/models/assets.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+use schema::assets;
+use utils::errors::ConvertToDatabaseError;
+use utils::errors::DatabaseError;
+use utils::errors::ErrorCode;
+use uuid::Uuid;
+
+#[derive(Queryable, Identifiable)]
+pub struct Asset {
+    pub id: Uuid,
+    blockchain_name: String,
```
Comment (SWvheerden) [inline db/src/models/assets.rs]: assuming this is the actual asset_id

File: db/src/models/assets.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    status: String,
+    created_at: NaiveDateTime,
+    updated_at: NaiveDateTime,
+}
+
+impl Asset {
+    pub fn create(blockchain_name: String) -> NewAsset {
+        NewAsset {
+            blockchain_name,
+            status: AssetStatus::Unsynced.to_string(),
```
Comment (SWvheerden) [inline db/src/models/assets.rs]: are we saving the data this was last sync'ed?

File: db/migrations/00000000000025_assets/up.sql (hunk 1 above)
Comment (mikethetike) [reply db/migrations/00000000000025_assets/up.sql]: I'm going to store both.

**Pull request merged**